### PR TITLE
Add order_by option to Query.changes/2

### PR DESCRIPTION
**Pull request by wkirschbaum**
Solves https://github.com/bitcrowd/carbonite/issues/113

I struggled to set up more than one change record for the test, so need some help there.

### CODE REVIEW

File: lib/carbonite/query.ex [hunk 1]
```diff
@@ -16,6 +16,7 @@ defmodule Carbonite.Query do
 
   @type prefix_option :: {:carbonite_prefix, prefix()}
   @type preload_option :: {:preload, boolean()}
+  @type order_by_option :: {:order_by, false | nil | tuple()}
```
Comment (maltoe) [inline lib/carbonite/query.ex:19]:
```suggestion
  @type order_by_option :: {:order_by, false | nil | term()}
```

Right? Ecto accepts keyword lists as well as tuples, right? And we don't even need to care, we can accept it as an opaque value in `maybe_order_by`:

```
value -> order_by(queryable, ^value)
```

File: test/carbonite/query_test.exs [hunk 2]
```diff
@@ -185,6 +185,10 @@ defmodule Carbonite.QueryTest do
       assert [%Change{data: %{"name" => "Jack"}}] = changes(rabbit)
     end
 
+    test "can disable ordering", %{rabbit: rabbit} do
+      assert Map.get(Query.changes(rabbit, order_by: false), :order_bys) == []
+    end
```
Comment (maltoe) [inline test/carbonite/query_test.exs:190]: This test is great and IMO the best option to assert on "query is not ordered"...

Maybe add another test for custom ordering. The rabbit fixtures inserted in `insert_rabbits` have fixed `age` values, for instance.

**Review by maltoe — CHANGES_REQUESTED** [review]
@wkirschbaum 

Thank you!

Could you add a line to top of the `CHANGELOG.md` as well?

```md
## Unreleased

### Added

* .... (@wkirschbaum)
```

File: test/carbonite/query_test.exs (hunk 2 above)
Comment (wkirschbaum) [reply test/carbonite/query_test.exs:190]: thanks, added this test

**Comment by wkirschbaum** [comment]
@maltoe thanks for the feedback and help. I updated the commit with the suggested changes I believe. Feel free to edit the commit if there is something small not to your liking.

**Pull request merged**